updater.go

```diff
@@ -184,6 +185,76 @@ func (u *Updater) update(ctx Context, options UpdateOptions) (*Update, error) {
 	return update, nil
 }
 
+func (u *Updater) ApplyDownloaded(ctx Context) error {
+	options := ctx.UpdateOptions()
+
+	// 1. check with the api server again for the latest update to be sure that a new update has not come out since our last call to CheckAndDownload
+	u.log.Infof("Attempting to apply previously downloaded update")
+	update, err := u.checkForUpdate(ctx, options)
+
+	if err != nil {
+		report(ctx, findErr(err), update, options)
+		return findErr(err)
+	}
+
+	err = u.applyDownloaded(ctx, update, options)
+
+	if err != nil {
+		report(ctx, err, update, options)
+		return err
+	}
+	return nil
+
+}
+
+// ApplyDownloaded will look for an previously downloaded update and attempt to apply it without prompting.
+// CheckAndDownload must be called first so that we have a download asset avaiable to apply.
+func (u *Updater) applyDownloaded(ctx Context, update *Update, options UpdateOptions) error {
+	if update == nil || !update.NeedUpdate {
+		u.log.Infof("No previously downloaded update to apply since client is update to date")
+		return nil
+	}
+	u.log.Infof("Got update with version: %s", update.Version)
+
+	if update.Asset == nil || update.Asset.URL == "" {
+		u.log.Info("No update asset to apply")
+		return nil
+	}
+
+	// 2. check the disk via FindDownloadedAsset. Compare our API result to this result. If the downloaded update is stale, clear it and start over.
+	downloadedAssetPath, err := u.FindDownloadedAsset(update.Asset.Name)
+
+	if err != nil {
+		return err
+	}
+
+	if downloadedAssetPath == "" {
+		u.log.Infof("No downloaded asset found for version: %s", update.Version)
+		if err := u.CleanupPreviousUpdates(); err != nil {
+			u.log.Infof("Error cleaning up previous downloads: %v", err)
+		}
+		// Bail
+		return nil
+	}
+	update.Asset.LocalPath = downloadedAssetPath
+
+	// 3. otherwise use the update on disk and apply it.
+	// Make sure that at the end of this flow (success or failure, we actually remove the downloaded update)
+	defer u.CleanupPreviousUpdates()
```

Comment:
I think you can set this right after downloading completes successfully. if we hit `downloadedAssetPath == ""` we want to run this anyway. Should log the error though